adapters/Chartboost/ChartboostAdapter/GADMAdapterChartboostSingleton.m

```diff
@@ -45,56 +36,20 @@ + (nonnull GADMAdapterChartboostSingleton *)sharedInstance {
   return sharedInstance;
 }
 
-- (nonnull instancetype)init {
-  self = [super init];
-  if (self) {
-    _completionHandlers = [[NSMutableArray alloc] init];
-  }
-  return self;
-}
-
 - (void)startWithAppId:(nonnull NSString *)appId
           appSignature:(nonnull NSString *)appSignature
      completionHandler:(nonnull ChartboostInitCompletionHandler)completionHandler {
-  switch (self->_initState) {
-    case GADMAdapterChartboostInitialized:
-      completionHandler(nil);
-      break;
-    case GADMAdapterChartboostInitializing:
-      GADMAdapterChartboostMutableArrayAddObject(self->_completionHandlers, completionHandler);
-      break;
-    case GADMAdapterChartboostUninitialized:
-      GADMAdapterChartboostMutableArrayAddObject(self->_completionHandlers, completionHandler);
-      self->_initState = GADMAdapterChartboostInitializing;
-
-      GADMAdapterChartboostSingleton *weakSelf = self;
       [Chartboost startWithAppId:appId
```

Comment:
Can you confirm the following regarding the `startWithAppId:` initialization method?

* Can it be called more than once and the completion handler is called synchronously every time ?
* Will it return immediately if Chartboost SDK has already been initialized?

Reply:
Of course, glad to help.
- Yes it can be called more than once, even while Chartboost is still starting. Each completion handler will be called, one at a time, on the main thread.
- And yes, it will return immediately, on the main thread. This means that if the startWithAppId: method is called on a background thread, the caller function may return before the completion has been executed on the main thread.

Reply:
Thank you for clarifying!